build/BuildSteps.cs

```diff
@@ -219,6 +219,32 @@ public static void BuildExtensionsBundle(BuildConfiguration buildConfig)
             }
         }
 
+        public static void GenerateVulnerabilityReport()
+        {
+            Settings.WindowsBuildConfigurations.ForEach((config) => RunVulnerabilityReport(config));
+        }
+
+        public static void RunVulnerabilityReport(BuildConfiguration buildConfig)
+        {
+            string projectDirectory = Path.Combine(Settings.RootBuildDirectory, buildConfig.ConfigId.ToString());
+            string projectFilePath = Path.Combine(Settings.RootBuildDirectory, projectDirectory, "extensions.csproj");
+
+            var currectDirectory = Directory.GetCurrentDirectory();

```

Comment:
Will there ever be the case where `Shell.GetOutput` would throw? If yes, we should use a finally block to ensure that currectDirectory is reset.

```
string currectDirectory = null;
try
{
    currectDirectory = Directory.GetCurrentDirectory();
   // run dotnet list...
}
finally
{
   // if currectDirectory is not null
   Directory.SetCurrentDirectory(currectDirectory);
}
```

Reply:
Not expecting Shell.GetOutput to throw but doesn't hurt to add some defensive logic.